### PR TITLE
Adds support for react-testing-library

### PR DESCRIPTION
**Pull request by kylekthompson**
Hello 👋 I recently added `react-testing-library` to a project I'm working on and wanted the ability to snapshot test some styles. I noticed https://github.com/styled-components/jest-styled-components/issues/129 was open, so I figured I'd give it a shot.

Let me know if there is anything else I should do here!

### CODE REVIEW

**Comment by kylekthompson** [comment]
Oh 🤦‍♂️ I'll also add some documentation to the README.

File: src/toHaveStyleRule.js [hunk 1]
```diff
@@ -18,6 +18,10 @@ const getClassNames = received => {
       if (components.length) {
         className = components.first().prop('className')
       }
+    } else if (received instanceof global.Element) {
+      // in react-testing-library, there is a container element
+      // around what was actually rendered, so we should grab the firstChild
+      className = Array.from(received.firstChild.classList).join(' ')
```
Comment (kylekthompson) [inline src/toHaveStyleRule.js]: Is this a reasonable assumption? It's meant to make it easy to do the following:

```js
const { container } = render(<Component />);
expect(container).toHaveStyleRule(...);
```

But, if we wanted to check a nested element, you'd have to wrap it in a container.

Comment (kylekthompson) [reply src/toHaveStyleRule.js]: After a little more reading, I'm leaning towards thinking that it's _not_ a reasonable assumption. Especially in the case when a fragment is rendered. From react-testing-library's README:

> NOTE: When that root element is a React Fragment, container.firstChild will only get the first child of that Fragment, not the Fragment itself.

^ that could lead towards some strange test behavior

**Comment by MicheleBertoli** [comment]
🙌 This is awesome, thank you very much @kylekthompson.
I love the minimalistic approach, and the documentation.

Overall LGTM and tests are green, so I'm happy to merge this.

Re: `firstChild`
I believe your assumption is correct, and the following test passes:

```js
test('fragment', () => {
  const Wrapper = styled.section`
    padding: 4em;
    background: papayawhip;
  `
  const { container } = render(
    <React.Fragment>
      <Wrapper />
    </React.Fragment>
  )

  expect(container).toHaveStyleRule('background', 'papayawhip')
})
```

File: README.md [hunk 2]
```diff
@@ -218,6 +219,21 @@ test('it works', () => {
...
+## react-testing-library
+
+To generate snapshots with [react-testing-library](https://github.com/kentcdodds/react-testing-library), you can just follow the example below:
+
+```js
+import { render } from 'react-testing-library'
+
+test('it works', () => {
+  const { container } = render(<Button />)
+  expect(container).toMatchSnapshot()
```
Comment (MicheleBertoli) [inline README.md]: I would change this to `container.firstChild` (as per `react-testing-library` [documentation](https://github.com/kentcdodds/react-testing-library#usage)).

File: src/toHaveStyleRule.js (hunk 1 above)
Comment (MicheleBertoli) [reply src/toHaveStyleRule.js]: > The alternative would be passing container.firstChild to expect

I think that's the way to go, as per `react-testing-library` [documentation](https://github.com/kentcdodds/react-testing-library#usage) on snapshot testing (and for consistency with the other matcher).

Comment (kylekthompson) [reply src/toHaveStyleRule.js]: Will do!

**Comment by kylekthompson** [comment]
@MicheleBertoli I went ahead and made a few updates to cover your comments.

1. Switched the expectations in the tests and documentation to use `container.firstChild`
1. Removed the `.firstChild` in `toHaveStyleRule` when building the class name
1. Updated the README to explain a bit better how to use `toHaveStyleRule` with `react-testing-library`

**Comment by MicheleBertoli** [comment]
perfect

**Pull request merged**